frontend/src/metabase/reference/reference.js

```diff
@@ -15,10 +16,14 @@ const END_LOADING = "metabase/reference/END_LOADING";
 export const endLoading = createAction(END_LOADING);
 
 const START_EDITING = "metabase/reference/START_EDITING";
-export const startEditing = createAction(START_EDITING);
+export const startEditing = createAction(START_EDITING, () => {
+    MetabaseAnalytics.trackEvent('Data Reference', 'Started Editing');
```

Comment:
Does this capture enough metadata to be useful? It might be a good idea to add `section.type` as a payload to the action to capture what we're editing at least.